src/vma/event/delta_timer.cpp

```diff
@@ -234,8 +236,9 @@ void timer::process_registered_timers()
 	while (iter && (iter->delta_time_msec == 0)) {
 		tmr_logfuncall("timer expired on %p", iter->handler);
 
-		if (iter->handler){
+		if (iter->handler && !iter->lock_timer.trylock()){
```

Comment:
Shouldn't we check that iter->handle != NULL, after we took the lock?

Reply:
I improved this.

Reply:
After thinking more my answer is "no" because `iter->handler` can not be updated after initial creation.

Reply:
what will happen if another thread calls remove timer?

Reply:
or this can't happen since the timer is removed only from inernal thread context, right?

Reply:
yep